app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java

```diff
@@ -144,6 +141,12 @@ public void feedMe(View view) {
         }
     }
 
+    public void excludeAllIngridients(View view) {
```

Comment:
You can call this method from clearIngredients to avoid code repetition

Reply:
I cannot use the function you wrote since the onPress function has to be a public function that accept a view as a parameter (first I tried using it, that's thy it is public